### PR TITLE
Plugins: Enable browsing the plugins repo on dotcom sites

### PR DESCRIPTION
**Pull request by dmsnell**
Includes: https://github.com/Automattic/wp-calypso/pull/9468

> <del>Needs to coordinate with other work to enable automatic transfer of
> sites. **Do not merge** until that time.</del>

> This is now properly hidden behind a feature flag

Previously when attempting to view the list of repo plugins on a dotcom
site the user would be redirected from `/plugins/browse/siteSlug` to
`/plugins/siteSlug` and shown the features included in WordPress.com
sites.

Now, the browser will be opened up for WordPress.com sites whereas in a
future PR the ability to install them will be activated too.

**Testing**
Navigate to a WordPress.com site in the plugins browser. In **master** it should redirect to the features list for WordPress.com sites but in this branch it should show the list from the plugins repository.

`/plugins/browse/mySiteSlug`

Try using the search and filtering options and make sure that it work as expected. Try switching between Jetpack and WordPress.com sites.

**Testing with `automated-transfer` enabled**
```bash
ENABLE_FEATURES=automated-transfer make run
```

**Testing with `automated-transfer` disabled**
```bash
DISABLE_FEATURES=automated-transfer make run
```

## Plugin Browser Routing
| Site Type | AT on? | Destination | Actual Result |
| --- | :-: | --- | :-: |
| Jetpack | N | plugin browser | ✔️ |
| Jetpack | Y | plugin browser | ✔️ |
| dotcom | N | dotcom features | Oops not supported |
| dotcom | Y | plugin browser | ✔️ |

## Single Plugin Routing
| Site Type | AT on? | Destination | Actual Result |
| --- | :-: | --- | :-: |
| Jetpack | N | plugin detail | ✔️ |
| Jetpack | Y | plugin detail | ✔️ |
| dotcom | N | dotcom features | ✔️ |
| dotcom | Y | plugin detail | ✔️ |

## Issues

 - When switching from the plugin browser on a dotcom site to a Jetpack site the back button routes to the list of installed plugins and not the plugin browser

cc: @roundhill @rodrigoi @vindl

### CODE REVIEW

File: client/my-sites/plugins/index.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ module.exports = function() {
 			)
 		) );
 
-		page( '/plugins/:plugin/:business_plugin?/:site_id?',
```
Comment (dmsnell) [inline client/my-sites/plugins/index.js]: left lingering after 4631897912a8d8876bf4a08d3fa905902f00f134?

cc: @Tug you think it's safe to remove this?

File: client/my-sites/plugins-wpcom/plugins-list.jsx [hunk 2]
```diff
@@ -1,35 +0,0 @@
-/**
```
Comment (dmsnell) [inline client/my-sites/plugins-wpcom/plugins-list.jsx]: We won't remove it but only for the feature flag. They should be updating a different file - `plugins-wpcom/index.jsx` or `plugins-wpcom/plugin-panel.jsx`

Comment (Copons) [inline client/my-sites/plugins-wpcom/plugins-list.jsx]: I've heavily changed `plugin-panel.jsx`, adding a whole new set of Jetpack-related components. Everything else in that folder is pretty much untouched.

File: client/my-sites/plugins/index.js (hunk 1 above)
Comment (Tug) [inline client/my-sites/plugins/index.js]: Yep it should be, I don't think it was kept on purpose either.

Comment (dmsnell) [inline client/my-sites/plugins/index.js]: Thanks @Tug

**Comment by vindl** [comment]
I've tried disabling the `automated-transfer` feature flag and I was still able to navigate to `/plugins/browse/siteSlug` for a dotcom site.

**Comment by Copons** [comment]
Do these routing make sense?

| Routing | Site Type | AT on? | Destination | Actual Result |
| --- | --- | :-: | --- | :-: |
| Plugin Browser | dotcom | N | dotcom features | Oops not supported |
| Single Plugin | dotcom | N | dotcom features | ✔️ |

I mean, it's not supported in both cases, but if anything it feels to me it would make more sense if they were the other way around:

- Plugin browser redirects to dotcom features (which is basically a plugin browser).
- Single plugin redirects to not supported.

File: client/my-sites/plugins/plugin-meta/index.jsx [hunk 3]
```diff
@@ -281,7 +296,7 @@ export default React.createClass( {
 
 		return (
 			<div className="plugin-meta">
-				<Card>
+				<Card style={ { margin: 0 } }>
```
Comment (roundhill) [inline client/my-sites/plugins/plugin-meta/index.jsx]: Let's add a className and move the styles to the CSS.

File: client/my-sites/plugins/plugin-meta/index.jsx [hunk 4]
```diff
@@ -293,15 +308,33 @@ export default React.createClass( {
...
+							site={ this.props.selectedSite }
+							pluginVersion={ plugin && plugin.version }
+							siteVersion={ this.props.selectedSite && this.props.selectedSite.options.software_version }
+							hasUpdate={ this.getAvailableNewVersions().length > 0 }
+						/>
+					}
 				</Card>
+
+				{ ( this.props.selectedSite.jetpack || this.hasBusinessPlan() ) &&
+					<div style={ { marginBottom: 16 } } />
```
Comment (roundhill) [inline client/my-sites/plugins/plugin-meta/index.jsx]: Should use class/css file here as well.

Comment (vindl) [inline client/my-sites/plugins/plugin-meta/index.jsx]: Updated.

File: client/lib/plans/constants.js [hunk 5]
```diff
@@ -477,6 +479,12 @@ export const FEATURES_LIST = {
 		)
 	},
 
+	[ FEATURE_UPLOAD_PLUGINS ]: {
+		getSlug: () => FEATURE_UPLOAD_PLUGINS,
+		getTitle: () => i18n.translate( 'Upload plugins' ),
```
Comment (roundhill) [inline client/lib/plans/constants.js]: Let's change this to `Install plugins`, and the description to `Install custom plugins on your site`.

Comment (vindl) [inline client/lib/plans/constants.js]: Sure, updated.

**Comment by roundhill** [comment]
:shipit:

**Pull request merged**